src/middleware/jirahost-middleware.ts

```diff
@@ -43,15 +43,14 @@ const detectJwtTokenType = (req: Request): TokenType => {
 export const jirahostMiddleware = async (req: Request, res: Response, next: NextFunction) => {
 
 	const unsafeJiraHost = extractUnsafeJiraHost(req);
-
 	req.addLogFields({ jiraHost: unsafeJiraHost });
 
 	// JWT validation makes sure "res.locals.jiraHost" is legit, not the cookie value. To avoid
 	// any temptation to use it later, let's remove it straight away!
 	const takenFromCookies = unsafeJiraHost === req.cookies.jiraHost;
 	res.clearCookie("jiraHost");
 
-	if (unsafeJiraHost) {
+	if (unsafeJiraHost && isDifferentFromSession(unsafeJiraHost, req.session.jiraHost)) {
```

Comment:
you don't need a separate function just do do `unsafeJiraHost !== res.session.jiraHost`

Reply:
true

Reply:
Also, I'm fairly certain we want to verify the JWT token on every request, even if the jirahost in the session hasn't changed because the token might have